v2/cf-nodejs-logging-support/src/lib/logger/record-factory.ts

```diff
@@ -0,0 +1,96 @@
+import util from "util";
+import Config from "../config/config";
+import NestedVarResolver from "../helper/nested-var-resolver";
+import RequestAccesor from "../middleware/request-accesor";
+import ResponseAccesor from "../middleware/response-accessor";
+import ReqContext from "./context";
+
+export default class RecordFactory {
+
+    // init a new record and assign fields with output "msg-log"
+    static buildMsgRecord(_args: Array<any>, _context?: ReqContext): any {
+
+        const msgLogFields = Config.getInstance().getMsgFields();
+        let record: any = {
+            "level": "info",
+        };
+
+        msgLogFields.forEach(field => {
+            if (!Array.isArray(field.source)) {
+                switch (field.source.type) {
+                    case "req-header":
+                        record[field.name] = _context?.getProp(field.name as string);
+                        break;
+                    case "req-object":
+                        record[field.name] = _context?.getProp(field.name as string);
+                        break;
+                    case "static":
+                        record[field.name] = field.source.value;
+                        break;
+                    case "env":
+                        if (field.source.path) {
+                            record[field.name] = NestedVarResolver.resolveNestedVariable(process.env, field.source.path);
+                            break;
+                        }
+                        record[field.name] = process.env[field.source.name!];
+                        break;
+                    case "config-field":
+                        record[field.name] = record[field.source.name!];
+                        break;
+                }
+            } else {
+
+                // TO DO: handle sources as array case
+            }
+
+        });
+        record["msg"] = util.format.apply(util, _args);
+        // TO DO: check if Stacktrace
+        return record;
+    }
+
+    // init a new record and assign fields with output "req-log"
+    static buildReqRecord(_req: any, _res: any): any {
+
+        const requestAccesor = RequestAccesor.getInstance();
+        const responseAccesor = ResponseAccesor.getInstance();
+
+        const reqLogFields = Config.getInstance().getReqFields();
+        let record: any = { "level": "info" };
+        reqLogFields.forEach(field => {
+            if (!Array.isArray(field.source)) {
+                switch (field.source.type) {
+                    case "req-header":
+                        record[field.name] = requestAccesor.getHeaderField(_req, field.source.name as string);
```

Comment:
Yes it is a type cast and it is in fact not necessary since we validate the imported config fields with a JSON Schema beforehand. I will eliminate them 👍 .